### PR TITLE
Updates for Ruby 2.3.0, 2.4.0 and RSpec

### PR DESCRIPTION
**Pull request by berlincount**
Not sure whether this is the PR you expected, but ...

@nerdrew did quite a bit of work for RSpec3 integration, which I cherry-picked

Also:

- added Ruby 2.3.0, 2.4.0
- removed Ruby ree, 1.8.7 Travis testing
- re-enabled some Travis tests
- moved a gem from 'development' to 'test' section

and tadaa, it builds ok for all reasonably current Ruby versions per https://travis-ci.org/berlincount/zk

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,11 +17,11 @@ before_install:
 rvm:
   - 1.9.3
   - 1.9.2
-  - 1.8.7
-  - ree
   - 2.0.0
   - 2.1.0
   - 2.2.0
+  - 2.3.0
+  - 2.4.0
```
Comment (nerdrew) [inline .travis.yml]: Should this be the most recent ruby versions (2.3.6 and 2.4.3)?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -61,31 +60,75 @@
       SpecGlobalLogger.logger.debug  { "stopping zookeeper service" }
       ZK::Server.shutdown
     end
+
+    config.before(:each, zookeeper: true) do
```
Comment (nerdrew) [inline spec/spec_helper.rb]: You don't need this whole change if you aren't testing throttling and lost connections. I accidentally left these changes in here. I used them in my gem https://github.com/nerdrew/ruby-zk_recipes to test lost connections and degraded connections.

File: .travis.yml (hunk 1 above)
Comment (berlincount) [reply .travis.yml]: sure, why not!

File: spec/spec_helper.rb (hunk 2 above)
Comment (berlincount) [reply spec/spec_helper.rb]: agreed. could maybe your extensions be worth merging to zk? I'm a fan of having fewer, better maintained gems, with more maintainers in general :)

**Comment by slyphon** [comment]
@eric any thoughts? not sure how you feel about ree & 1.8.7 getting dropped

File: spec/spec_helper.rb (hunk 2 above)
Comment (nerdrew) [reply spec/spec_helper.rb]: I'm happy to merge my cache recipe into this gem if the maintainers are cool with it. That was my original intention, but got no responses on the issues I opened.

Comment (nerdrew) [reply spec/spec_helper.rb]: In fact, I'd like to make some changes to logging in this gem. With lots of watches, a ZK restart took our app down b/c of the ZK gem logging volume.

**Comment by nerdrew** [comment]
Any movement on this?

**Comment by slyphon** [comment]
Hey sorry man, I talked w/ @eric, and he gave his blessing.

File: spec/spec_helper.rb (hunk 2 above)
Comment (slyphon) [reply spec/spec_helper.rb]: Hey, so, to give you a bit of the backstory, @eric and I worked on this together back in like 2011 or thereabouts. I was using it for a task queuing system and he was using it for his company. I took the original zookeeper repo over from Twitter, and added zk on top of it.

I haven't _really_ touched the code since 2013 (though it's still in active use in parts of twitter). @eric was keeping on top of things because he was still using it actively, until he sold his company a year ago and has been kinda just enjoying life.

Basically what I'm saying is, I haven't been an active user of this codebase for a long time. Hell, I haven't written any *ruby* in over 4 years. If you guys are using zk in production, and are willing to put in the work to do releases and maintenance of this, then by all means, I'm happy to help out however I can (including if that means getting out of the way).

Comment (nerdrew) [reply spec/spec_helper.rb]: Want to add some people to the contributor list so we can review PRs, release versions, etc?

Comment (nerdrew) [reply spec/spec_helper.rb]: I'd like to merge my cache recipe (https://github.com/nerdrew/ruby-zk_recipes) into this gem  but I'm worried about not being able to make changes quickly.

**Comment by slyphon** [comment]
sure, lemme take a look on Monday. We'll probably have to rework the
release stuff so it's "modern". Last time i touched it was 5 years ago, and
I'm sure there's better stuff out there than my hand-rolled deploy scripts
:)
